assets/src/settings-page/template-modes.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * External dependencies
+ */
+import {
+	THEME_IS_SUPPORTED,
+	THEME_SUPPORT_ARGS,
+	THEME_SUPPORTS_READER_MODE,
+} from 'amp-settings';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { TemplateModeOption } from '../components/template-mode-option';
+import '../css/template-mode-selection.css';
+import { AMPNotice, NOTICE_SIZE_LARGE, NOTICE_TYPE_INFO, NOTICE_TYPE_WARNING } from '../components/amp-notice';
+import { Options } from '../components/options-context-provider';
+
+/**
+ * Provides the notice to show in the reader theme support mode selection.
+ *
+ * @param {string} themeSupport The current theme support mode.
+ */
+function getReaderNotice( themeSupport ) {
+	switch ( true ) {
+		// Theme has built-in support or has declared theme support with the paired flag false.
+		case 'reader' === themeSupport && ( THEME_IS_SUPPORTED || ( 'object' === typeof THEME_SUPPORT_ARGS && false === THEME_SUPPORT_ARGS.paired ) ):
```

Comment:
I think the `THEME_IS_SUPPORTED` check should be removed here:

```suggestion
		case 'reader' === themeSupport && ( ( 'object' === typeof THEME_SUPPORT_ARGS && false === THEME_SUPPORT_ARGS.paired ) ):
```

This is because the core theme sanitizer allows for the themes to work in both AMP and non-AMP (so they work in standard or transitional).

Reply:
Just a reminder (for myself too):

* If a theme has `paired:false` then technically we should just say that it works in Standard mode. Otherwise,
* If a theme has `paired:true` (or no `paired` defined at all) then it works in Standard mode _or_ Transitional mode.

We should change `AMP_Theme_Support::get_theme_support_args()`  so that if `paired` is not defined, it defaults to `false`. This should be the default behavior. In practice, all AMP-compatible themes really need to support Transitional mode because a site can turn off AMP for certain templates. When that happens, they need to provide non-AMP functionality.

So we really are only concerned with when a theme says `paired:false` because that's a signal that they really don't want a theme to be used to generate non-AMP pages. But again, in practice this is probably extremely uncommon, and it may not even be something we need to be concerned about.

So really, as long as `THEME_SUPPORT_ARGS` is not `false`, that's all we really care about.

Reply:
Updated in 6e83093936d78865233c2bdf1087284393fec557